### PR TITLE
Doc/security handler

### PR DESCRIPTION
**Pull request by Jiehong**
Fixes warning on request.rst at doc generation time, and inline the example for the security with basic auth.

### CODE REVIEW

File: docs/security.rst [hunk 1]
```diff
@@ -56,10 +56,18 @@ semantics as for ``x-tokenInfoFunc``, but the function accepts three
 parameters: username, password and required_scopes. If the security declaration
 of the operation also has an oauth security requirement, required_scopes is
 taken from there, otherwise it's None. This allows authorizing individual
-operations with oauth scope while using basic authentication for
+operations with `oauth scope`_ while using basic authentication for
 authentication.
 
-You can find a `minimal Basic Auth example application`_ in Connexion's "examples" folder.
+You can find a `minimal Basic Auth example application`_ in Connexion's "examples" folder:
```
Comment (scoulomb) [inline docs/security.rst:62]: maybe we could also point to the readme of the example

File: docs/security.rst [hunk 2]
```diff
@@ -56,10 +56,18 @@ semantics as for ``x-tokenInfoFunc``, but the function accepts three
...
+
+
+.. literalinclude:: ../examples/openapi3/basicauth/app.py
```
Comment (scoulomb) [inline docs/security.rst:65]: In the example I am wondering if  this line https://github.com/zalando/connexion/blob/bed4b95205205861bd6014282d0f61d50231d3ac/examples/openapi3/basicauth/app.py#L31 makes sense?

Comment (Jiehong) [reply docs/security.rst]: The example was showing the oauth2 as well, so I fixed that by improving the documentation and linking to the correct example for oauth2, and I removed that dummy_func function as well, as it no longer makes sense.

File: docs/security.rst (hunk 1 above)
Comment (Jiehong) [reply docs/security.rst]: Yes, you're right. I removed the inline code, and used the link instead.

**Comment by Jiehong** [comment]
@szborows : hi, what do you think of this PR?

**Comment by jmcs** [comment]
:+1:

**Pull request merged**